File: NeuralNetworks-python/NeuralNetworks/src/_math/MathUtils.py
'''
Created on 21 de set de 2019

@author: marcelovca90
'''
import numpy as np
import random as rnd

class MathUtils:
    
    def __init__(self):
        pass

def mean_squared_error(w, x, d):
    mse = 0
    for i in range(0,len(x)):
        v = np.dot(np.transpose(w), x[i])
        mse = mse + pow(d[i] - v, 2)
    mse = mse / len(x)
    return mse

File: NeuralNetworks-java/src/io/github/marcelovca90/nn/math/MathUtils.java
package io.github.marcelovca90.nn.math;

public class MathUtils
{
    public static double dotProduct(double[] d1, double[] d2)
    {
        double ans = 0.0;
        for (int i = 0; i < d1.length; i++)
            ans += (d1[i] * d2[i]);
        return ans;
    }

    public static double meanSquaredError(double[] d1, double[] d2)
    {
        double ans = 0.0;
        for (int i = 0; i < d1.length; i++)
            ans += Math.pow(d1[i] - d2[i], 2.0);
        ans /= d1.length;
        return ans;
    }
}
